src/main/java/com/google/sps/api/calendar/CalendarClientAdapter.java

```diff
@@ -108,7 +108,19 @@ public ZonedDateTime getUsersTomorrowStart() {
         .collect(Collectors.toList());
   }
 
+  /**
+   * Inserts the event in the primary calendar.
+   */
   public void insertEventToPrimary(Event event) throws IOException {
     calendarClient.events().insert(PRIMARY_CALENDAR_FLAG, event).execute();
```

Comment:
this might be quite inefficient for many events. Even though there is no support for bulk insert, there seems to be support for batching requests. (part of it has been deprecated)
Maybe file an issue to investigate general performance of the scheduler operations. This might be some interesting optimization to work on towards the end of the project.